credentials/alts/internal/handshaker/handshaker.go

```diff
@@ -43,7 +43,7 @@ const (
 	rekeyRecordProtocolName = "ALTSRP_GCM_AES128_REKEY"
 	// maxPendingHandshakes represents the maximum number of concurrent
 	// handshakes.
-	maxPendingHandshakes = 100
```

Comment:
There's an important behavior difference between this max concurrency limit and C-core's: in Go, it looks like we fail handshakes fast if we would otherwise surpass the concurrency limit. Meanwhile in C-core, we enqueue such handshakes.

I think this fail-fast behavior makes lowering the concurrency limit here somewhat risky (servers that were handling a certain client handshake QPS before may see an increased rate of error).

Maybe before reducing this limit, we should change the Go behavior to queue.

Reply:
Sure thing, I changed it back to 100 for this PR.